pkg/dns/dns.go

```diff
@@ -19,42 +19,66 @@ import (
 	"fmt"
 	"os"
 	"regexp"
+	"strings"
 )
 
 const (
 	// Default location for the resolv.conf file.
 	DefaultResolveConfPath = "/etc/resolv.conf"
 
 	// Default cluster domain value for k8s clusters.
-	DefaultLocalDNS = "svc.cluster.local"
+	DefaultClusterDomain = "cluster.local"
 )
 
-// GetLocalDNSName parses the path to resolv.conf to find the local DNS name.
-func GetLocalDNSName(resolvConfPath string) (string, error) {
-	var localDNSName string
+// GetClusterDomain parses the path to resolv.conf to find the cluster domain.
+func GetClusterDomain(resolvConfPath string) (string, error) {
+	var clusterDomain string
 	file, err := os.Open(resolvConfPath)
 	if err != nil {
 		return "", err
 	}
 	defer file.Close()
 
-	reg := regexp.MustCompile(`^search.*?\s(svc\.[^\s]*)`)
+	reg := regexp.MustCompile(`^search.*?\ssvc\.([^\s]*)`)
 
 	scanner := bufio.NewScanner(file)
 	for scanner.Scan() {
 		match := reg.FindStringSubmatch(scanner.Text())
 		if len(match) > 0 {
-			localDNSName = match[1]
+			clusterDomain = match[1]
 		}
 	}
 
 	if err := scanner.Err(); err != nil {
 		return "", err
 	}
 
-	if localDNSName == "" {
-		return "", fmt.Errorf("failed to find local DNS name in resolv.conf")
+	if clusterDomain == "" {
+		return "", fmt.Errorf("failed to find cluster domain in resolv.conf")
 	}
 
-	return localDNSName, nil
+	return clusterDomain, nil
+}
+
+// GetServiceDNSNames parses the fully-qualified domain service name nd
+// returns a list of its qualified service names.
+func GetServiceDNSNames(fqdnServiceName string, clusterDomain string) ([]string, error) {
+	all := strings.Split(fqdnServiceName, ".")
```

Comment:
Feels weird that we're doing a `fmt.Sprintf(` to build `fqdnServiceName` where we're calling this, and then immediately splitting it here and validating a certain number of elements.

I think this function would be more user-friendly if it just took the service name and namespace as separate string arguments instead of as a combined fqdn.

Reply:
If you're okay with it I think I'd like to leave this as-is and refactor in a follow-up PR I need to make

Reply:
Fixed in 6ade567